### PR TITLE
Product listing

### PR DESCRIPTION
**Pull request by KaustubhM1997**
Fetched data from the server
refactored some CSS
Created context for the product page

Live - https://xenodochial-sammet-bf3770.netlify.app/product-listing

please note - the aside bar is not functional yet.

### CODE REVIEW

File: src/main.css [hunk 1]
```diff
@@ -21,6 +21,9 @@ body {
   --badge-background: black;
   --button-hover: #be185d;
   --background-color: #fdf2f8;
+  --product-discount: #22c55e;
+  --text-line-through: gray;
+  --card-box-shadow: lightgray;
```
Comment (mimansha-swarup) [inline src/main.css:26]: good practice

File: src/pages/Home/home.css [hunk 2]
```diff
@@ -105,7 +102,8 @@
   background-color: var(--primary-color);
   color: white;
   overflow: hidden;
-  position: sticky;
+  /* position: static; */
+  /* bottom: 0; */
```
Comment (mimansha-swarup) [inline src/pages/Home/home.css]: try to remove unnecessary comments

File: src/pages/Product-listing/product-listing.css [hunk 3]
```diff
@@ -1,5 +1,203 @@
 @import url(../../main.css);
 
-.title {
-  margin-top: 8rem;
+.main-wrapper {
+  display: grid;
+  min-height: 100vh;
+  grid-gap: 2rem;
+  /* The aside will take 1fr and the rest (content) will take 7fr */
```
Comment (mimansha-swarup) [inline src/pages/Product-listing/product-listing.css:7]: its great that you are documenting it too

File: src/pages/Product-listing/product-listing.js [hunk 4]
```diff
@@ -1,16 +1,40 @@
-import "../Home/home.css";
 import React from "react";
-import { Nav } from "../../components/Nav/nav";
+import "../Home/home.css";
 import "./product-listing.css";
-import { Footer } from "../../components/Footer/footer";
+import { Aside } from "../../components/Aside/aside";
+import { useProduct } from "../../contexts/product-context";
+import { ProductCard } from "../../components/Product-card/product-card";
```
Comment (mimansha-swarup) [inline src/pages/Product-listing/product-listing.js:6]: try using barrel export

**Review by mimansha-swarup — APPROVED** [review]
I loved how you are  commenting in between code
it really increase code redability

**Review by Shrinkhlarajpoot — COMMENTED** [review]
Use barrel export if possible and good folder structure.

**Review by Shrinkhlarajpoot — APPROVED** [review]

**Pull request merged**